### PR TITLE
(WIP) Support multiform/file submission stubbing

### PR DESCRIPTION
**Pull request by gavintmyers**
For the first time we are now attaching files to form submissions / fetches /
POSTs. In order to test this, we need to be able to mock and stub such fetches.
File-containing submissions take the form of a "multipart/formdata" request rather than
a "urlencoded" request. More details can be found here:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST When a multipart
request contains a binary file in its body (such as a png), this can cause
encoding difficulties. In our case, runner.rb uses JSON.pretty_generate() on
the entire result of every dump, which includes that binary-containing body,
causing an error when .pretty_generate() cannot convert it to UTF-8 encoding as
it requires.

The chosen solution has changed the dump output structure such that request and response
bodies are both always given their own separate file, .request_body and .response_body.
These are written with default UTF-8 encoding, in a way that is safe to run on possible binaries, but also
should keep the more common, nonbinary bodies easily readable in a text editor.

### CODE REVIEW

File: lib/rails_response_dumper/runner.rb [hunk 1]
```diff
@@ -119,6 +117,12 @@ def run_dumps
             dump[:timestamp] = timestamp.iso8601 unless options[:exclude_timestamp]
 
             File.write("#{dumper_dir}/#{index}.json", JSON.pretty_generate(dump))
+            File.open("#{dumper_dir}/#{index}.request_body", 'w', external_encoding: 'ASCII-8BIT') do |f|
```
Comment (jdufresne) [inline lib/rails_response_dumper/runner.rb:120]: You can use `File.write` for a shorter, more concise syntax:

The body is binary data and shouldn't go through a text encoding.

I also think we should drop "_body" from these file names as I don't find it adds any context but it does add general noise to the directory. 

Same for the response.

```suggestion
            File.write("#{dumper_dir}/#{index}.request", request.body.string, mode: 'wb')
```

File: package.json
Comment (jdufresne) [inline package.json:1]: This file looks unrelated, please remove it.

File: package-lock.json
Comment (jdufresne) [inline package-lock.json:1]: This file looks unrelated, please remove it.

**Review by jdufresne — COMMENTED** [review]
Please update the README so it remains accurate.

Please update the changelog so we can immediately prep a release.

I don't see a new test added. I think we should add one that shows binary data in a POST that models the use case.

**Comment by jdufresne** [comment]
For a bit of technical accuracy: multiform -> multipart/form-data

**Comment by gavintmyers** [comment]
Closing and deleting branch to use a fork instead
